.github/workflows/ci.yml

```diff
@@ -0,0 +1,133 @@
+name: CI
+
+on:
+  pull_request:
+    paths:
+      - "workspaces/**"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  detect-workspaces-with-changes:
+    name: Detect workspaces with changes
+    runs-on: ubuntu-latest
+    outputs:
+      workspaces: ${{ steps.detect-workspace.outputs.out }}
+    steps:
+      - name: Checkout main branch for diff purposes
+        uses: actions/checkout@v4
+        with:
+          ref: main
+      - name: Checkout head branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.workflow_run.head_branch }}
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20.x"
+      - name: Install root dependencies
+        run: yarn install --immutable
+      - name: Set workspaces with changes as output
+        id: detect-workspace
+        run: echo "out=$(yarn run detect-workspace)" >> $GITHUB_OUTPUT
+
+  ci:
+    name: Run CI for workspace ${{ matrix.workspace }} using node ${{ matrix.node-version }}
```

Comment:
```suggestion
    name: Workspace ${{ matrix.workspace }}, CI step for node ${{ matrix.node-version }}
```

I think if you name them like this, you'll get the verify + ci right next to each other which probably is easier to navigate in the list. Let me know if that assumption is wrong.

Also should Verify be above CI? Just a matter of taste but feels kinda logical and probably finishes much faster too so the green dot is above the busy one